test/__snapshots__/test/test.ts.md

```diff
@@ -447961,15 +447871,78 @@ Generated by [AVA](https://avajs.dev).
             nohoist?: string[];␊
             [k: string]: unknown;␊
           };␊
-      jspm?: JSONSchemaForNPMPackageJsonFiles1;␊
+      jspm?: JSONSchemaForNPMPackageJsonFiles;␊
       /**␊
        * Any property starting with _ is valid.␊
        *␊
-       * This interface was referenced by \`JSONSchemaForNPMPackageJsonFiles2\`'s JSON-Schema definition␊
+       * This interface was referenced by \`undefined\`'s JSON-Schema definition␊
```

Comment:
This seems to have regressed

Reply:
Just a heads up: I haven't lost interest in moving this forward. I've had some deadlines come up, but I plan to return to this over the weekend.

Reply:
Hello! Thank you for your patience. I've taken a look at this regression and I'm not sure what the expected behavior should be. Could you advise?

This is what I observe, simplified to the relevant bits.

Given this schema:

```yaml
title: JSON schema for NPM package.json files
type: object
patternProperties:
  '^_':
    description: Any property starting with _ is valid.
properties:
  # etc, etc, etc, actual properties clipped
```

The original code produced this structure:

```ts
export type JSONSchemaForNPMPackageJsonFiles =
  JSONSchemaForNPMPackageJsonFiles1 & JSONSchemaForNPMPackageJsonFiles2;

export type JSONSchemaForNPMPackageJsonFiles1 = { [k: string]: unknown; };

export interface JSONSchemaForNPMPackageJsonFiles2 {␊
  /**
   * Any property starting with _ is valid.
   * 
   * This interface was referenced by \`JSONSchemaForNPMPackageJsonFiles2\`'s
   * JSON-Schema definition via the \`patternProperty\` "^_".
   */
  [k: string]: any;
}
```

This branch produces a different structure:

```ts
export type JSONSchemaForNPMPackageJsonFiles =
  { [k: string]: unknown; } &
  {
    /**
     * Any property starting with _ is valid.
     * 
     * This interface was referenced by \`undefined\`'s JSON-Schema definition
     * via the \`patternProperty\` "^_".
     */
    [k: string]: any;
  };
```

While I agree that the change in the comment from `JSONSchemaForNPMPackageJsonFiles2` to `undefined` is unfortunate, this behavior is in line with other instances in which this comment is generated due to the presence of `patternProperties` on the root schema.

For instance, here is the current output of the `intersection.2.ts` test on the `master` branch, which has a similar scenario:

```ts
export type Intersection = (A & B) & {
  c: string;
  d: string;
  /**
   * This interface was referenced by \`undefined\`'s JSON-Schema definition
   * via the \`patternProperty\` "^x-".
   */
  [k: string]: unknown;
};

export interface A {
  a?: string;
  [k: string]: unknown;
}

export interface B {
  b?: string;
  [k: string]: unknown;
}
```

I would argue that neither behavior is ideal, and an improvement might be to suppress the comment entirely if the interface is not named.

I can make this change, but I worry that it may not be what you have in mind, and that it feels like something that should be a separate PR.

Could you please provide direction on how this might be resolved?

Reply:
That seems reasonable -- we can improve this as a follow-up.